website/content/en/docs/component-guides/katib.md

```diff
@@ -6,197 +6,265 @@ weight = 20
-           description: Training model optimizer (sdg, adam or ftrl)
-           reference: optimizer
-       trialSpec:
-         apiVersion: batch/v1
-         kind: Job
-         spec:
-           template:
-             metadata:
-               annotations:
-                 sidecar.istio.io/inject: "false"
-             spec:
-               containers:
-                 - name: training-container
-                   image: public.ecr.aws/z1j2m4o4/kubeflow-katib-mxnet-mnist:latest
-                   command:
-                     - "python3"
-                     - "/opt/mxnet-mnist/list_s3_buckets.py"
-                     - "&&"
-                     - "python3"
-                     - "/opt/mxnet-mnist/mnist.py"
-                     - "--batch-size=64"
-                     - "--lr=\${trialParameters.learningRate}"
-                     - "--num-layers=\${trialParameters.numberLayers}"
-                     - "--optimizer=\${trialParameters.optimizer}"
-               restartPolicy: Never
-   EOF
-   ```
-
-10. Create the experiment.
+#### 2. `katib-config` config map configuration
+
+The [`katib-config`](https://www.kubeflow.org/docs/components/katib/katib-config/) contains configurations involving metrics collection, tuning algorithms, and early stopping algorithms.
+
+By default, pods that will run the tuning (suggestion) algorithm are created under the `default` service account present in the profile namespace. However, the `AwsIamForServiceAccount` plugin annotates the `default-editor` service account with the profile's `awsIamRole`, which means that only pods created under the `default-editor` service account will be granted the desired AWS permissions.
+
+The below steps will modify the `katib-config` to create pods under the `default-editor` service account so that the pods will be granted the desired permissions.
+
+1. Open the `katib-config` config map for editing.
+    ```bash
+    kubectl edit configMap katib-config -n kubeflow
+    ```
+
+2. Navigate to the `suggestion` volume settings. The settings will look as follows:
+    ```yaml
+    suggestion: |-
+    {
+      "random": {
+        "image": "docker.io/kubeflowkatib/suggestion-hyperopt",
+        ...
+      },
+      "tpe": {
+        "image": "docker.io/kubeflowkatib/suggestion-hyperopt:v0.13.0",
+        ...
+      }
+      ...
+    }
+    ```
+
+3. For each algorithm (e.g `random`, `tpe`, etc.) add a key for `serviceAccountName` with a value of `default-editor`:
+    ```yaml
+    suggestion: |-
+    {
+      "random": {
+        "image": "docker.io/kubeflowkatib/suggestion-hyperopt",
+        "serviceAccountName": "default-editor"
+        ...
+      },
+      "tpe": {
+        "image": "docker.io/kubeflowkatib/suggestion-hyperopt:v0.13.0",
+        "serviceAccountName": "default-editor"
+        ...
+      }
+      ...
+    }
+    ```
+
+4. Close the edit window. This will apply the configuration.
+
+#### 3. Experiment Trial Spec configuration
```

Comment:
Are both 2 and 3 needed or can we just do this configuration? If 3 is sufficiant, lets decouple both and keep the focus on configuring trial spec and the sample for this first since I expect this to be most widely used.

for 2 lets just have a section after trial spec things giving a few details and maybe link it to here: https://www.kubeflow.org/docs/components/katib/katib-config/#suggestion-settings for reasons:
- we dont have a use case yet
- I think the algorithm pod might just be used to control what trials are created
- the configmap(deployment/add-ons/profile-aws-iam-plugin/patch_katib_config_map_sa.yaml) will need to be updated every release which is ok but I dont see a strong reason to do it because of the first reason 😄

Reply:
done